core/element_unary_operations.h

```diff
@@ -44,6 +45,66 @@ constexpr auto sqrt_out_arg =
                  x = sqrt(y);
                }};
 
+constexpr auto sin_out_arg = overloaded{
+    arg_list<double, float>, transform_flags::expect_no_variance_arg<0>,
+    transform_flags::expect_no_variance_arg<1>, [](auto &x, const auto y) {
+      using std::sin;
+      x = sin(y);
+    }};
+
+constexpr auto cos_out_arg = overloaded{
+    arg_list<double, float>, transform_flags::expect_no_variance_arg<0>,
+    transform_flags::expect_no_variance_arg<1>, [](auto &x, const auto y) {
+      using std::cos;
+      x = cos(y);
+    }};
+
+constexpr auto tan_out_arg = overloaded{
+    arg_list<double, float>, transform_flags::expect_no_variance_arg<0>,
+    transform_flags::expect_no_variance_arg<1>, [](auto &x, const auto y) {
+      using std::tan;
+      x = tan(y);
+    }};
+
+constexpr auto asin =
+    overloaded{transform_flags::expect_no_variance_arg<0>, [](const auto x) {
+                 using std::asin;
+                 return asin(x);
+               }};
+
+constexpr auto asin_out_arg = overloaded{
+    arg_list<double, float>, transform_flags::expect_no_variance_arg<0>,
+    transform_flags::expect_no_variance_arg<1>, [](auto &x, const auto y) {
+      using std::asin;
+      x = asin(y);
+    }};
+
+constexpr auto acos =
+    overloaded{transform_flags::expect_no_variance_arg<0>, [](const auto x) {
+                 using std::acos;
```

Comment:
@SimonHeybrock I think I just ended up copying this style from elsewhere in scipp, but is there a reason for the `using` rather than just `return std::acos(x)`?

Reply:
Yes, there is a reason: Argument-dependent lookup. `std::acos` does not support `Unit` or `ValueAndVariance`, so we use ADL to find the correct overload. The `using` brings the overload from `std` into scope.